x-pack/platform/packages/shared/kbn-classic-stream-flyout/src/components/create_classic_stream_flyout/create_classic_stream_flyout.tsx

```diff
@@ -46,29 +50,99 @@ const flyoutBodyStyles = css({
 });
 
 interface CreateClassicStreamFlyoutProps {
+  /** Callback when the flyout is closed */
   onClose: () => void;
-  onCreate: () => void;
+  /**
+   * Callback when the stream is created.
+   * Receives the stream name which can be used to create the classic stream.
+   */
+  onCreate: (streamName: string) => void;
+  /** Callback to navigate to create template flow */
   onCreateTemplate: () => void;
+  /** Available index templates to select from */
   templates: TemplateDeserialized[];
-  selectedTemplate: string | null;
-  onTemplateSelect: (templateName: string | null) => void;
+  /** Whether there was an error loading templates */
   hasErrorLoadingTemplates?: boolean;
+  /** Callback to retry loading templates */
   onRetryLoadTemplates: () => void;
+  /**
+   * Async callback to validate the stream name.
+   * Called after local empty field validation passes.
+   * Should check for duplicate names and higher priority template conflicts.
+   */
+  onValidate?: StreamNameValidator;
 }
 
 export const CreateClassicStreamFlyout = ({
   onClose,
   onCreate,
   onCreateTemplate,
   templates,
-  selectedTemplate,
-  onTemplateSelect,
   hasErrorLoadingTemplates = false,
   onRetryLoadTemplates,
+  onValidate,
 }: CreateClassicStreamFlyoutProps) => {
   const [currentStep, setCurrentStep] = useState<ClassicStreamStep>(
     ClassicStreamStep.SELECT_TEMPLATE
   );
+  const [selectedTemplate, setSelectedTemplate] = useState<string | null>(null);
+  const [streamName, setStreamName] = useState<string>('');
+  const [selectedIndexPattern, setSelectedIndexPattern] = useState<string>('');
+  const [validationError, setValidationError] = useState<ValidationErrorType>(null);
+  const [conflictingIndexPattern, setConflictingIndexPattern] = useState<string | undefined>();
+  const [hasAttemptedSubmit, setHasAttemptedSubmit] = useState(false);
+  const [isValidating, setIsValidating] = useState(false);
+
+  const selectedTemplateData = templates.find((t) => t.name === selectedTemplate);
+
+  // Run validation and update state, returns true if validation passes
+  const runValidation = useCallback(
+    async (name: string): Promise<boolean> => {
+      setIsValidating(true);
+      try {
+        const result = await validateStreamName(name, onValidate);
```

Comment:
if this fails, we are silently swallowing the error since we dont have a catch. Is this something we should worry about?